cpp/src/arrow/engine/substrait/serde.h

```diff
@@ -39,38 +39,38 @@ namespace engine {
 /// each toplevel Substrait relation when deserializing a Substrait Plan.
 using ConsumerFactory = std::function<std::shared_ptr<compute::SinkNodeConsumer>()>;
 
-/// \brief Deserializes a Substrait Plan message to a list of ExecNode declarations
+/// \brief Deserializes a Substrait Plan message to a list of ExecNode declarations.
 ///
 /// The output of each top-level Substrait relation will be sent to a caller supplied
-/// consumer function provided by consumer_factory
+/// consumer function provided by consumer_factory.
 ///
 /// \param[in] buf a buffer containing the protobuf serialization of a Substrait Plan
-/// message
+/// message.
 /// \param[in] consumer_factory factory function for generating the node that consumes
-/// the batches produced by each toplevel Substrait relation
+/// the batches produced by each toplevel Substrait relation.
 /// \param[in] registry an extension-id-registry to use, or null for the default one.
 /// \param[out] ext_set_out if non-null, the extension mapping used by the Substrait
 /// Plan is returned here.
 /// \return a vector of ExecNode declarations, one for each toplevel relation in the
-/// Substrait Plan
+/// Substrait Plan.
 ARROW_ENGINE_EXPORT Result<std::vector<compute::Declaration>> DeserializePlans(
     const Buffer& buf, const ConsumerFactory& consumer_factory,
     const ExtensionIdRegistry* registry = NULLPTR, ExtensionSet* ext_set_out = NULLPTR);
 
-/// \brief Deserializes a single-relation Substrait Plan message to an execution plan
+/// \brief Deserializes a single-relation Substrait Plan message to an execution plan.
 ///
-/// The output of each top-level Substrait relation will be sent to a caller supplied
-/// consumer function provided by consumer_factory
+/// The output of each top-level Substrait relation will be sent to the given consumer
+/// node.
```

Comment:
```suggestion
/// The output of each top-level Substrait relation will be sent to the given consumer.
```
Minor nit: A `SinkNodeConsumer` isn't actually an exec node.  An exec node has to worry about various lifecycle concerns but a consumer just receives batches.